tools/clang/tools/dxclib/dxc.cpp

```diff
@@ -803,10 +803,8 @@ void DxcContext::Recompile(IDxcBlob *pSource, IDxcLibrary *pLibrary,
                            pIncludeHandler, &pResult));
   }
 
-#ifndef _WIN32
-  // FIXME: fix crash on linux when not detach pCompileSource.
+  // Detach pCompileSource because the data is owned by pPdbUtils.
```

Comment:
I believe the proper fix is in the loop earlier - to change `pCompileSource.Attach(pSourceFile);` to `pCompileSource = pSourceFile` or `pCompileSource.Attach(pSourceFile.Detach())` to transfer the reference.

Reply:
Updated to pCompileSource.Attach(pSourceFile.Detach())